### PR TITLE
Parse app detail based on script tags data

### LINKED ISSUE #192: TypeError: Cannot read property 'split' of undefined
version: 5.0.0

invode param:
```javascript
gPlay.list({
        category : gPlay.category.GAME,
        collection : gPlay.collection.NEW_FREE,
        lang : 'zh',
        country : 'CN',
        fullDetail : true,
        start : 0,
        num : 100,
    }).then(function (appList) {
        //console.log(appList);
        cb(null, appList);
    }).catch(function (err) {
        console.error("try to spider googleplay app list failed, err=", err);
        cb(err);
    });

```

log:
```
at parseFields (/googleplay/node_modules/google-play-scraper/lib/app.js:48:64)
    at tryCatcher (/googleplay/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/googleplay/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/googleplay/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/googleplay/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/googleplay/node_modules/bluebird/js/release/promise.js:693:18)
    at Async._drainQueue (/googleplay/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/googleplay/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/googleplay/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:794:20)
    at tryOnImmediate (timers.js:752:5)
    at processImmediate [as _immediateCallback] (timers.js:729:5)
```

### PR DESCRIPTION
**Pull request by facundoolano**
This fixes #192 by extracting the app data from the js code in the script tags of the app detail html.

All the script data is added to a map object (with keys like `ds:13`, `ds:8`, etc). A `MAPPINGS` constant declares how to extract each field from that parsed object.

Only added the first couple of fields so far, feel free to PR this branch adding new fields to `MAPPINGS`.

- [x] tests passing
- [x] migrate important fields
- [x] add a test for preregister apps (search "pre register"+ fullDetail)
- [x] add a test for apps with no downloads (list last page of top new paid + fullDetail)
- [x] create issues for missing fields
- [x] update readme to reflect new fields (both app.js and list.js)

### CODE REVIEW

**Comment by EasyGitHub99** [comment]
I'm not sure how to contribute properly, still learning things but I wanted to help out:

`  developerAddress: ['ds:3', 0, 12, 5, 4, 0],

  histogram1: ['ds:14', 0, 1, 1, 1],
  histogram2: ['ds:14', 0, 1, 2, 1],
  histogram3: ['ds:14', 0, 1, 3, 1],
  histogram4: ['ds:14', 0, 1, 4, 1],
  histogram5: ['ds:14', 0, 1, 5, 1],
  histogramTotal: ['ds:14', 0, 2, 1],`

Is it proper github etiquette to edit your pull request to add these in, or do I create my own pull request for you to accept/reject?

Just wanted to help you out to get some of the values.

File: lib/app.js [hunk 1]
```diff
@@ -30,167 +28,209 @@ function app (opts) {
...
-  }
-
-  const characters = [' - ', ' et ', '–', '-', '～', ' a '];
-  const char = characters.find((char) => downloads.split(char).length === 2);
-  if (char) return downloads.split(char);
 
-  throw new Error(`Unable to parse min/max downloads: ${downloads}`);
+// TODO may need to do html to text in some of the fields
+
+const MAPPINGS = {
```
Comment (icaroponce) [inline lib/app.js]: Why don't we keep this `MAPPINGS` object clean, with only the paths? I think we could use a function to transform the data later, for example using the `R.evolve` with a different object for the fields that need some kind of transformation. The way it's done now makes difficult to express object values, like the `histogram`, which has this as paths: 

```
histogram: {
	1:['ds:10', 0 , 1, 1, 1],
	2:['ds:10', 0 , 1, 2 ,1],
	3:['ds:10', 0 , 1, 3 ,1],
	4:['ds:10', 0 , 1, 4 ,1],
	5:['ds:10', 0 , 1, 5 ,1],  
}
```
I can't simply pass this object without changing the logic that the `extractData` handles the spec when it is an object. This `extractData` wouldn't need to do smth much different when we got an object, simply get the path to it's properties.

So I believe we could do something like: 

```
const tranformData = R.evolve({
  developerId: (devUrl) => devUrl.split('id=')[1],
  score: parseFloat,
  ...
});
...then(matchScriptTag).then(extractData).then(transformData)
```

**Comment by icaroponce** [comment]
I also have some fields here:

```
  size: ['ds:5', 0],
  version: ['ds:5', 1],
  androidVersion: ['ds:5', 2]
  score: ['ds:10', 0 , 0, 0],
  price: ['ds:8',0 , 2, 0, 0, 0, 1, 0, 2],
  histogram: {
    1:['ds:10', 0 , 1, 1, 1],
    2:['ds:10', 0 , 1, 2 ,1],
    3:['ds:10', 0 , 1, 3 ,1],
    4:['ds:10', 0 , 1, 4 ,1],
    5:['ds:10', 0 , 1, 5 ,1],
  }
```

`score` still needs to be parsed to float, since it comes as string, and `histogram` would need changes as mentioned above.

**Comment by facundoolano** [comment]
@EasyGitHub99 the most convenient to me is for you to create a PR against this branch, so I can merge them together. But if you want to just post the mappings here it's also helpful, since I can just copy them next time I sit to update this branch.

**Comment by EasyGitHub99** [comment]
I think they're in ds:14
`  score: ['ds:14', 0, 0, 0],`
`  scoreFloat: ['ds:14', 0, 0, 1],`

**Comment by facundoolano** [comment]
that ds:14 mapping doesn't seem correct @EasyGitHub99

**Comment by facundoolano** [comment]
it's ['ds:10', 0 , 0, 0] and ['ds:10', 0 , 0, 1]

**Comment by facundoolano** [comment]
(I'm starting to worry that the ds:10, ds:14 thing is not always the same :S )

File: lib/app.js (hunk 1 above)
Comment (icaroponce) [reply lib/app.js]: I've just seen your solution for that, it seems good! Thanks 👍

**Comment by icaroponce** [comment]
@facundoolano I know the `preregister` already and I'm pushing a new PR is some hours once I'm back home. There's also some bugs that I caught here, that happens when the app is "`preregister`", it doesn't have `price` nor `minInstalls` for example... If you test with this app `com.gamedevltd.destinywarfare` for example, the scraper crashes.

**Comment by facundoolano** [comment]
Merged. Thanks @icaroponce and @EasyGitHub99 for your collaborations!

**Pull request merged**